libs/dal/src/repositories/subscriber/subscriber.schema.ts

```diff
@@ -165,11 +163,14 @@ subscriberSchema.index({
  *    subscriberId: /on-boarding-subscriber/i,
  *  });
  */
-subscriberSchema.index({
-  subscriberId: 1,
-  _environmentId: 1,
-  _id: 1,
-});
+subscriberSchema.index(
+  {
+    subscriberId: 1,
+    _environmentId: 1,
+    _id: 1,
```

Comment:
I think the `_id` part here can create duplicates still, as the id will always be random and unique. 

Mongodb already have a unique id index by default, So I think it should be 👍 
```
 subscriberId: 1,
    _environmentId: 1,
```

Wdyt?

Reply:
Good point, i think the main trade off here is if we want full coverage on the query or reduce memory, this is a big collection so I believe it will be significant.
looking at it again I wonder if mongo db would perform Index Prefix Intersection with both 
{subscriberId: 1, _environmentId: 1} and {_id} indexes.